edb/schema/functions.py

```diff
@@ -970,6 +970,24 @@ def get_verbosename(
         sn = self.get_shortname(schema)
         return f"function '{sn}{params.as_str(schema)}'"
 
+    def get_dummy_body(self, schema: s_schema.Schema) -> expr.Expression:
+        rt = self.get_return_type(schema)
+
+        if rt.is_scalar():
+            # scalars and enums can be cast from a string
+            text = f'SELECT <{rt.get_displayname(schema)}>""'
+        elif rt.is_object_type():
+            # just grab an object of the appropriate type
+            text = f'SELECT {rt.get_displayname(schema)} LIMIT 1'
+        else:
+            # Can't easily create a valid cast, so just cast empty set
+            # into the given type. Technically this potentially breaks
+            # cardinality requirement, but since this is a dummy
+            # expression it doesn't matter at the moment.
+            text = f'SELECT <{rt.get_displayname(schema)}>{{}}'
```

Comment:
Maybe this is the only branch that's actually needed?

Reply:
Not really. We cannot return an empty set from a function that is not returning an `OPTIONAL`, and most scalar functions are like that.  But the `else` branch is not entirely correct either, it should be `SELECT {rt.get_displayname(schema)} LIMIT 1`

Reply:
Actually the `else` branch is about collections.